src/Microsoft.Azure.WebJobs.Host/Bindings/FunctionBinding.cs

```diff
@@ -25,7 +25,9 @@ public FunctionBinding(FunctionDescriptor descriptor, IReadOnlyDictionary<string
         {
             Dictionary<string, IValueProvider> results = new Dictionary<string, IValueProvider>();
 
-            BindingContext bindingContext = new BindingContext(context, null);
+            // Supplied bindings can be direct parameters or route parameters. 
```

Comment:
Also, I'm a little concerned about overloading this dictionary so that it contains both raw argument values (the current behavior) as well as binding data (the new behavior you're adding).